app/bolt_listeners.py

```diff
@@ -91,41 +94,54 @@ def respond_to_app_mention(
             return
 
         user_id = context.actor_user_id or context.user_id
-
-        if payload.get("thread_ts") is not None:
+        if thread_ts is not None:
             # Mentioning the bot user in a thread
             replies_in_thread = client.conversations_replies(
                 channel=context.channel_id,
-                ts=payload.get("thread_ts"),
+                ts=thread_ts,
                 include_all_metadata=True,
                 limit=1000,
             ).get("messages", [])
             for reply in replies_in_thread:
                 reply_text = redact_string(reply.get("text"))
-                messages.append(
+                content = [
                     {
-                        "role": (
-                            "assistant"
-                            if "user" in reply and reply["user"] == context.bot_user_id
-                            else "user"
-                        ),
-                        "content": (
-                            f"<@{reply['user'] if 'user' in reply else reply['username']}>: "
-                            + format_openai_message_content(
-                                reply_text, TRANSLATE_MARKDOWN
-                            )
-                        ),
+                        "type": "text",
+                        "text": f"<@{reply['user'] if 'user' in reply else reply['username']}>: "
+                        + format_openai_message_content(reply_text, TRANSLATE_MARKDOWN)
                     }
-                )
+                ]
+
+                # Add images if any
+                files = reply.get(files)
+                if files:
```

Comment:
files are not always image ones. you can move this if statement into the append_image_content; renaming the method append_image_content_if_exists would be clearer in the case